mintpy/bulk_plate_motion.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/env python3
+############################################################
+# Program is part of MintPy                                #
+# Copyright (c) 2013, Zhang Yunjun, Heresh Fattahi         #
+# Author: Yuan-Kai Liu, May 2022                           #
+############################################################
+#
+# Extra dependency:
+#   + platemotion: https://github.com/lcx366/PlateTectonic
+#   + astropy
+#   How to install both:
+#      option (1) pip install platemotion
+#      option (2) git clone git@github.com:lcx366/PlateTectonic.git $TOOL_DIR/PlateTectonic
+#                 echo 'export PYTHONPATH=$PYTHONPATH:$TOOL_DIR/PlateTectonic' >> ~/.bashrc
+#                 somehow install other dependencies in setup.py using your conda
+#
+# To-Do List (updated 2022.5.30 Yuan-Kai Liu):
+#   + Replace scipy.interpolate with alternatives for efficiency. E.g.:
+#       skimage.resize https://scikit-image.org/docs/stable/auto_examples/transform/plot_rescale.html
+#       check mintpy usage https://github.com/insarlab/MintPy/blob/7adb3a11f875b832488a0c8e44c174d98b1df254/mintpy/tropo_gacos.py#L129
+#   + Calculate Euler rotation to multi-points ENU motion is slow (called by pmm2enu_at() here)
+#       In `platemotion` package, use array operation rather than for loops (at https://github.com/lcx366/PlateTectonic/blob/main/platemotion/classes/plate.py#L153)
+#   + Replace platemotion package by equations of Euler trasnformation to relax this dependency at all?
+
+import os
+import sys
+import scipy
+import pyproj
+import argparse
+import numpy as np
+
+from mintpy.utils import readfile, writefile, utils as ut
+from mintpy.diff import diff_file
+from mintpy.save_gmt import get_geo_lat_lon
+from mintpy.solid_earth_tides import prepare_los_geometry
+
+# https://docs.astropy.org/en/stable/units/index.html
+try:
+    from platemotion import Plate
+    from astropy import units as u
+except ImportError:
+    msg = 'Can NOT import platemotion!'
+    msg += '\nCheck more details at https://github.com/lcx366/PlateTectonic.'
+    raise ImportError(msg)
+
+
+
+#########################################  Usage  ##############################################
+REFERENCE = """reference:
+  Stephenson, O. L., Liu, Y. K., Yunjun, Z., Simons, M., Rosen, P. and Xu, X., (2022), The Impact of
+    Plate Motions on Long-Wavelength InSAR-Derived Velocity Fields, Geophys. Res. Lett. (under review)
+    doi:10.1002/essoar.10511538.2
+  Peter, H., Fernández, M., Aguilar, J., & Fernández, J. (2021). Copernicus POD Product Handbook: 
+    Copernicus Sentinel-1, -2 and -3 Precise orbit Determination Serivice (CPOD) (GMV-CPOD-TN-0009). 
+    https://sentinels.copernicus.eu/documents/247904/3372484/Sentinels-POD-Product-Handbook-1.19.pdf
+"""
```

Comment:
Please add 1) a brief description and 2) a complete reference for each of the 3 plate motion models (NNR-NUVEL1A, NNR-MORVEL56, NNR-ITRF14) here, as we discussed offline, to help users to decide which model they may want to use.